File: 0242-valid-anagram/solution.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function (s, t) {
    const map1 = {}
    if (s.length != t.length) return false
    for (let i = 0; i < s.length; i++) {
        const char = s[i]
        if (!map1[char]) map1[char] = 0
        map1[char] = map1[char] + 1
    }
    for (let i = 0; i < t.length; i++) {
        const char = t[i]
        if (map1[char] <= 0 || !map1[char]) {
            return false
        } else {
            map1[char] = map1[char] - 1
        }
    }
    return true

};

File: 0242-valid-anagram/solution.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }

        unordered_map<char, int> s_map;

        for (int i = 0, n = s.size(); i < n; i++) {
            if (s_map.find(s[i]) != s_map.end()) {
                s_map[s[i]]++;
            } else {
                s_map.insert({s[i], 1});
            }
        }

        for (int i = 0, n = s.size(); i < n; i++) {
            if (s_map.find(t[i]) != s_map.end()) {
                if (s_map[t[i]] <= 0) {
                    return false;
                } else {
                    s_map[t[i]]--;
                }
            } else {
                return false;
            }
        }

        return true;
    }
};